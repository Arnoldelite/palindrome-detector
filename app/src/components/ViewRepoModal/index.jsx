import React from 'react'
import { Modal, Card, Avatar, Tag, Badge, Icon } from 'antd'
import { observer } from 'mobx-react'
import Particles from 'react-particles-js'
import { store } from '../..'
import './style.scss'

const { Meta } = Card

@observer
class ViewRepoModal extends React.Component {
  render() {
    return (
      <Modal
        className="edit-modal"
        destroyOnClose
        title={`${store.repoInfo.language} Popular Repo info`}
        onCancel={() => store.showViewRepoModal(false)}
        visible={store.isViewRepoModalVisible}
        footer={null}
      >
        <Card
          hoverable
          style={{ width: 400 }}
          actions={[
            <Tag color="green"> stars: {store.repoInfo.stargazers_count}</Tag>,
            <Tag color="green"> watchers: {store.repoInfo.watchers}</Tag>,
            <Tag color="green"> issues: {store.repoInfo.open_issues}</Tag>,
          ]}
        >
          <Meta
            avatar={
              <Avatar
                src={store.repoInfo.owner !== undefined ? store.repoInfo.owner.avatar_url : null}
              />
            }
            title={<a href={store.repoInfo.html_url}>{store.repoInfo.full_name}</a>}
            description={store.repoInfo.description}
          />
        </Card>
        <Particles
          className="particles"
          params={{
            particles: {
              number: {
                value: 50,
              },
              size: {
                value: 3,
              },
            },
            interactivity: {
              events: {
                onhover: {
                  enable: true,
                  mode: 'repulse',
                },
              },
            },
          }}
        />
      </Modal>
    )
  }
}

export default ViewRepoModal
