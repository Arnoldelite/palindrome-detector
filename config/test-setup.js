const shell = require('shelljs')

module.exports = async () => {
  try {
    await shell.exec('npm run mongodb-parse')
  } catch (e) {
    console.log(e)
  }
}
