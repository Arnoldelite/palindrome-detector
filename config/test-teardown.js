const shell = require('shelljs')

module.exports = async () => {
  try {
    await shell.exec('docker-compose kill')
  } catch (e) {
    console.log(e)
  }
}
