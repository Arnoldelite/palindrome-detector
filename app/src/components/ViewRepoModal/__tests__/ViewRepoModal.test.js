import React from 'react'

// import Parse from 'parse/node'
import { shallow } from 'enzyme'

import ViewRepoModal from '..'
import { shallowToJson } from 'enzyme-to-json'
import '../../../../../config/jest/enzyme.config'
// import { expect } from 'chai';
// import sinon from 'sinon';

// import MyComponent from './MyComponent'
// import Foo from './Foo'

describe('<MyComponent />', () => {
  test('renders <ViewRepoModal /> component', () => {
    const wrapper = shallow(<ViewRepoModal />)
    console.log('test component repo modal >>', wrapper)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
