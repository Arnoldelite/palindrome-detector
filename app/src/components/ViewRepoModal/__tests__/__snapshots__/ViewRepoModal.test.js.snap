// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyComponent /> renders <ViewRepoModal /> component 1`] = `
<Modal
  cancelButtonDisabled={false}
  className="edit-modal"
  confirmLoading={false}
  destroyOnClose={true}
  footer={null}
  maskTransitionName="fade"
  okButtonDisabled={false}
  okType="primary"
  onCancel={[Function]}
  title="undefined Popular Repo info"
  transitionName="zoom"
  visible={false}
  width={520}
>
  <Card
    actions={
      Array [
        <Tag
          closable={false}
          color="green"
        >
           stars: 
        </Tag>,
        <Tag
          closable={false}
          color="green"
        >
           watchers: 
        </Tag>,
        <Tag
          closable={false}
          color="green"
        >
           issues: 
        </Tag>,
      ]
    }
    hoverable={true}
    style={
      Object {
        "width": 400,
      }
    }
  >
    <_default
      avatar={
        <Avatar
          shape="circle"
          size="default"
          src={null}
        />
      }
      title={<a />}
    />
  </Card>
  <t
    className="particles"
    height="100%"
    params={
      Object {
        "interactivity": Object {
          "events": Object {
            "onhover": Object {
              "enable": true,
              "mode": "repulse",
            },
          },
        },
        "particles": Object {
          "number": Object {
            "value": 50,
          },
          "opacity": Object {
            "anim": Object {
              "enable": true,
              "opacity_min": 0.5,
              "speed": 2,
            },
            "value": 0.5,
          },
          "size": Object {
            "value": 3,
          },
        },
      }
    }
    style={Object {}}
    width="100%"
  />
</Modal>
`;
